libcxx/test/libcxx/language.support/support.c.headers/support.c.headers.other/stddefneeds.verify.cpp

```diff
@@ -0,0 +1,185 @@
+static void f2(void) { unreachable(); } // expected-error{{undeclared identifier}}
+max_align_t m2;                         // expected-error{{unknown type}}
+size_t o2 =
+    offsetof(struct astruct, member); // expected-error{{expected expression}} expected-error{{undeclared identifier}}
+wint_t wi2;                           // expected-error{{unknown type}}
+
+#define __need_rsize_t
+#include <stddef.h>
+
+ptrdiff_t p3;
+size_t s3;
+rsize_t r3;
+wchar_t wc3;
+void* v3 = NULL;                        // expected-error{{undeclared identifier}}
+nullptr_t n3;                           // expected-error{{unknown type}}
+static void f3(void) { unreachable(); } // expected-error{{undeclared identifier}}
+max_align_t m3;                         // expected-error{{unknown type}}
+size_t o3 =
+    offsetof(struct astruct, member); // expected-error{{expected expression}} expected-error{{undeclared identifier}}
+wint_t wi3;                           // expected-error{{unknown type}}
+
+#define __need_wchar_t
+#include <stddef.h>
+
+ptrdiff_t p4;
+size_t s4;
+rsize_t r4;
+wchar_t wc4;
+void* v4 = NULL;                        // expected-error{{undeclared identifier}}
+nullptr_t n4;                           // expected-error{{unknown type}}
+static void f4(void) { unreachable(); } // expected-error{{undeclared identifier}}
+max_align_t m4;                         // expected-error{{unknown type}}
+size_t o4 =
+    offsetof(struct astruct, member); // expected-error{{expected expression}} expected-error{{undeclared identifier}}
+wint_t wi4;                           // expected-error{{unknown type}}
+
+#define __need_NULL
+#include <stddef.h>
+
+ptrdiff_t p5;
+size_t s5;
+rsize_t r5;
+wchar_t wc5;
+void* v5 = NULL;
+nullptr_t n5;                           // expected-error{{unknown type}}
+static void f5(void) { unreachable(); } // expected-error{{undeclared identifier}}
+max_align_t m5;                         // expected-error{{unknown type}}
+size_t o5 =
+    offsetof(struct astruct, member); // expected-error{{expected expression}} expected-error{{undeclared identifier}}
+wint_t wi5;                           // expected-error{{unknown type}}
+
+// nullptr_t doesn't get declared before C23 because its definition
+// depends on nullptr.
+#define __need_nullptr_t
+#include <stddef.h>
+
+ptrdiff_t p6;
+size_t s6;
+rsize_t r6;
+wchar_t wc6;
+void* v6 = NULL;
+nullptr_t n6;
+static void f6(void) { unreachable(); } // expected-error{{undeclared identifier}}
+max_align_t m6;                         // expected-error{{unknown type}}
+size_t o6 =
+    offsetof(struct astruct, member); // expected-error{{expected expression}} expected-error{{undeclared identifier}}
+wint_t wi6;                           // expected-error{{unknown type}}
+
+#define __need_unreachable
+#include <stddef.h>
+
+ptrdiff_t p7;
+size_t s7;
+rsize_t r7;
+wchar_t wc7;
+void* v7 = NULL;
+nullptr_t n7;
+static void f7(void) { unreachable(); }
```

Comment:
Should this be allowed? Should the clang version of stddef.h never declare the `unreachable` macro even if `__need_unreachable` is set? `__need_unreachable` is new in LLVM 18.

Reply:
I'm pretty sure this is going to be a problem. https://github.com/llvm/llvm-project/pull/86748